gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsProjectSettings.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.components.Storage;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.Nullable;
+
+@com.intellij.openapi.components.State(
+        name = "ProjectSettings",
+        storages = {@Storage("gcv-plugin-settings.xml")})
+@Service(Service.Level.PROJECT)
+public final class VersionPropsProjectSettings implements PersistentStateComponent<VersionPropsProjectSettings.State> {
+
+    public enum EnabledState {
+        UNKNOWN,
+        ENABLED,
+        DISABLED
+    }
+
+    public static final class State {
+        // Using an enum like this ensure that the file is created in .idea allowing for end users to configure it for
+        // all users
+        private EnabledState enabled = EnabledState.UNKNOWN;
```

Comment:
We were going to get rid of this, right?

Reply:
Okay I found a happy middle ground - if I make the state as below then when enabled is false a file is created - this also means that a user can manually create the file to set enabled to false
```
    public static final class State {
        private boolean enabled = true;

        public void setEnabled(boolean enabled) {
            this.enabled = enabled;
        }

        public boolean getEnabled() {
            return this.enabled;
        }
    }
```